address.md

```diff
@@ -0,0 +1,262 @@
+    // 1: test-net
+    network byte
+    
+    // 0: ID
+    // 1: Blake2b160-Hash of secp256k1 Public Key
+    // 2: Blake2b160-Hash of Actor creation data
+    // 3: BLS Public Key
+    protocol byte
+    
+    // raw bytes containing the data associated with protocol
+    payload []byte
+}
+```
+
+#### Network Indicator
+
+The **network** byte identifies the network the address belongs to:
+
+- `0` : Mainnet
+- `1` : Testnet
+
+An example description in golang:
+
+```go
+// String Prefix
+Mainnet_Prefix = 'f'
+Testnet_Prefix = 't'
+
+type Protocol = byte
+const (
+    Mainnet Network = iota
+    Testnet
+)
+```
+
+#### Protocol Indicator
+
+The **protocol indicator** byte describes how a method should interpret the information in the payload field of an address. Any deviation for the algorithms and data types specified by the protocol must be assigned a new protocol number. In this way, protocols also act as versions.
+
+- `0`: ID
+- `1` : SECP256K1 Public Key
+- `2` : Actor
+- `3` : BLS Public Key
+
+An example description in golang:
+
+```go
+// Protocol byte
+type Protocol = byte
+const (
+    ID Protocol = iota
+    SECP256K1
+    Actor
+    BLS
+)
+```
+
+###### Protocol 0: IDs
+
+**Protocol 0** addresses are simple ids. These addresses are expected to remain within the network. All actors have a numeric ID even if they don't have public keys. The key part of an ID address is the decimal representation of the id only characters '0'-'9'. IDs are not hashed and do not have a checksum.
+
+###### Protocol 1: libsecpk1 Elliptic Curve Public Keys
+
+**Protocol 1** addresses represent secp256k1 public encryption keys. The payload field contains the [Blake2b 160](https://blake2.net/) hash of the public key. 
+
+###### Protocol 2: Actor
+
+**Protocol 2** addresses representing an Actor. The payload field contains the [Blake2b 160](https://blake2.net/) hash of meaningful data produced as a result of creating the actor. 
+
+###### Protocol 3: BLS
+
+**Protocol 3** addresses represent BLS public encryptions keys. The payload filed contains the BLS public key.
+
+#### Payload
+
+The payload represents the data specified by the protocol. All payloads excecpt the payload of the ID protocol are base32 encoded when seralized to their human readable format.
+
+#### Checksum
+
+The checksum is calculated using the first 4 bytes of the blake2b-160 hash of the addresses network, protocol, and payload bytes. Checksums are only used when seralizing and deseralizing an address to its human readable format.
+
+
+#### Expected Methods
+
+All implementations in Filecoin must have methods for creating, encoding, and decoding addresses in addition to checksum creation and validation. The follwing is a golang version of the Address Interface:
+
+```go
+type Address interface {
+    New(n byte, p byte, payload []byte) Address
+    Encode(a Adress) string
+    Decode(s string) Address
+    Checksum(a Address) []byte
+    Validate(a Address) bool
+}
+```
```

Comment:
no error handling?

Reply:
We use `Fatal` when there is an error, basically a panic. Validation has been added here: https://github.com/filecoin-project/specs/pull/135/files#diff-c6ea617359921b8070df6dd947c8af6cR257

Reply:
New is a method on an address that returns an address?

Reply:
moved it off the interface